src/main/java/io/pakland/mdas/githubstats/shell/components/CommitComponent.java

```diff
@@ -0,0 +1,34 @@
+package io.pakland.mdas.githubstats.shell.components;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.shell.command.CommandRegistration;
+import org.springframework.shell.standard.ShellComponent;
+import org.springframework.shell.standard.ShellMethod;
+import org.springframework.shell.standard.ShellOption;
+
+@ShellComponent
+public class CommitComponent {
+
+    public void commits(String user) {
+        System.out.println("commits");
+    }
+
+    @Bean
```

Comment:
Differences between using `@ShellCommand` (which I have seen in the docs) vs `@Bean`? Not really aware, just a question to understand it better.

Reply:
As I understand it, `@Bean` is just the way we would mark the `CommandRegistration` so that it can operate within Spring logic as mentioned in [Section 4.2](https://docs.spring.io/spring-shell/docs/2.1.2/site/reference/htmlsingle/#_programmatic_model). 

Bean definition by Spring docs: _In Spring, the objects that form the backbone of your application and that are managed by the Spring IoC container are called beans. A bean is an object that is instantiated, assembled, and otherwise managed by a Spring IoC container._

We need `CommandRegistration` to register CLI commands. The alternatives are `@ShellMethod`s and `@ShellOption`s but (quote): _The @ShellMethod and @ShellOption annotations are a legacy feature that we do not yet want to remove. CommandRegistration is the new development model where new features are added. We are most likely going to replace existing annotations with something better, to support new features in a CommandRegistration model._

Not sure what is `@ShellCommand` - do you mean `@ShellCommandGroup`?

Reply:
I meant `@ShellMethod`...

Reply:
Ok, so then this part explains it _The @ShellMethod and @ShellOption annotations are a legacy feature [...] CommandRegistration is the new development model where new features are added._

We could potentially work with any of them but choosing the most up-to-date solution seems reasonable.

Reply:
Ofc, no need ot use deprecated features.